h/storage.py

```diff
@@ -187,9 +184,16 @@ def update_annotation(session, id_, data):
     # annotation object.
     document = data.pop('document', None)
 
-    annotation = session.query(models.Annotation).get(id_)
+    annotation = request.db.query(models.Annotation).get(id_)
     annotation.updated = updated
 
+    group = group_service.find(annotation.groupid)
+    if group is None:
+        raise schemas.ValidationError('group: ' +
+                                      _('Invalid group specified for annotation'))
+    if request.feature('filter_groups_by_scope') and data.get('target_uri', None):
```

Comment:
Note: This logic means that an annotation that may have made its way into the db at a previous time with an invalid `target_uri` (per scope constraints on its `group`) will be able to be re-saved/updated—scope validation here will only occur if the request contains a `target_uri` field. One hopes there are zero instances of this but I thought I'd point it out.

Reply:
This sounds like the right choice to me.

Reply:
I think the main time this would be an issue is when adding or removing scopes from the group, which is a thing we have not currently implemented.

Reply:
@robertknight 

> I think the main time this would be an issue is when adding or removing scopes from the group, which is a thing we have not currently implemented.

Is this worth tracking as a separate issue (data integrity for annotations as group scopes are added, changed or deleted)?